src/frontend/src/lib/components/signer/SignerOrigin.svelte

```diff
@@ -0,0 +1,45 @@
+<script lang="ts">
+	import type { Origin, PayloadOrigin } from '@dfinity/oisy-wallet-signer';
+	import { isNullish, nonNullish } from '@dfinity/utils';
+	import ExternalLink from '$lib/components/ui/ExternalLink.svelte';
+	import { i18n } from '$lib/stores/i18n.store';
+	import type { Option } from '$lib/types/utils';
+
+	export let payload: Option<PayloadOrigin>;
+
+	let origin: Origin | undefined;
+	$: origin = payload?.origin;
+
+	const mapHost = (origin: Origin | undefined): Option<string> => {
```

Comment:
there is an `OptionString` type already built, IIRC

Reply:
`Origin | undefined` comes from the OISY Wallet Signer library. This is correct.

```
export const UrlSchema = z
  .string()
  .url()
  .refine(
    (url): boolean => {
      try {
        const {protocol, hostname} = new URL(url);

        // We allow http for development locally
        if (['localhost', '127.0.0.1'].includes(hostname)) {
          return ['http:', 'https:'].includes(protocol);
        }

        return protocol === 'https:';
      } catch (_err: unknown) {
        return false;
      }
    },
    {
      message: 'Invalid URL.'
    }
  );
  
export const OriginSchema = UrlSchema;

const PayloadOriginSchema = z.object({
  origin: OriginSchema
});
```

Reply:
What i meant is that the return is `Option<string>`, but we have a type already called `OptionString` that is that

Reply:
Ahhhhhhhhhhhhhhhhh

Reply:
Thx. Done!